CRM/Core/Form/Task/PDFLetterCommon.php

```diff
@@ -198,7 +198,19 @@ public static function setDefaultValues() {
    */
   public static function formRule($fields, $files, $self) {
     $errors = [];
-    $template = CRM_Core_Smarty::singleton();
+    $deprecatedTokens = [
+      '{case.status_id}' => '{case.status_id:label}',
+      '{case.case_type_id}' => '{case.case_type_id:label}',
+    ];
+    $tokenErrors = [];
+    foreach ($deprecatedTokens as $token => $replacement) {
+      if (strpos($fields['html_message'], $token) !== FALSE) {
+        $tokenErrors[] = ts('Token %1 is no longer supported - use %2 instead', ['{case.case_type_id}', '{case.case_type_id:label}']);
```

Comment:
Oh just noticed this always says case_type_XXX. Should be the loop vars.

Reply:
opps